subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/ForkingTestClassProcessor.java

```diff
@@ -45,11 +47,13 @@
     private final Iterable<File> classPath;
     private final Action<WorkerProcessBuilder> buildConfigAction;
     private final ModuleRegistry moduleRegistry;
+    private final Lock lock = new ReentrantLock();
     private RemoteTestClassProcessor remoteProcessor;
     private WorkerProcess workerProcess;
     private TestResultProcessor resultProcessor;
     private WorkerLeaseRegistry.WorkerLeaseCompletion completion;
     private DocumentationRegistry documentationRegistry;
+    private volatile boolean stoppedNow;
```

Comment:
⭕️ Doesn't need to be volatile if you're synchronizing access.

Reply:
Done ... the same thing occurred to me last night after I had signed off for the day.  I was planning on making this change first thing this morning.